Extension/src/LanguageServer/ui_new.ts

```diff
@@ -416,14 +417,27 @@ export class NewUI implements UI {
     }
 
     private showConfigureIntelliSenseButton: boolean = false;
+    private cppHasBeenActive: boolean = false;
+
     public async ShowConfigureIntelliSenseButton(show: boolean, client?: Client): Promise<void> {
         if (!telemetry.showStatusBarIntelliSenseButton() || client !== this.currentClient) {
             return;
         }
         if (show) {
             this.showConfigureIntelliSenseButton = true;
-            this.configureIntelliSenseStatusItem.show();
+            const activeEditor: vscode.TextEditor | undefined = vscode.window.activeTextEditor;
             telemetry.logLanguageServerEvent('configureIntelliSenseStatusBar');
+            if (activeEditor) {
+                const isCpp: boolean = activeEditor.document.uri.scheme === "file" && (activeEditor.document.languageId === "c" || activeEditor.document.languageId === "cpp" || activeEditor.document.languageId === "cuda-cpp");
```

Comment:
Create a util function for `isCpp` and `isCppPropertiesJson` since the check is used more than once.

Reply:
sure

Reply:
I feel like this was added in another PR, but we didn't take it yet. https://github.com/microsoft/vscode-cpptools/pull/10559/files#diff-1c5f3d6b951c5bd69dc09ef8a4bfa18807bb18d572579c2675f82a8a853284f7